app/models/event.rb

```diff
@@ -86,4 +100,28 @@ def compute_free_places
   def compute_occupied_places
     application_letters.where(status: ApplicationLetter.statuses[:accepted]).count
   end
+  
+  protected
+  def compare_participants_by_agreement(participant1, participant2)
+    if participant1.older_than_18_at_start_date_of_event?(self)
```

Comment:
Consider changing the function name to something like "needs_agreement_letter_for_event?", since that is what you actually want to know.